src/BenchmarksServer/Startup.cs

```diff
@@ -230,34 +234,94 @@ public static int Main(string[] args)
+                                else
+                                {
+                                    var benchmarksDir = await CloneRestoreAndBuild(tempDir, job, dotnetHome);
 
-                                Debug.Assert(process == null);
-                                process = StartProcess(hostname, database, sqlConnectionString, Path.Combine(tempDir, benchmarksDir),
-                                    job, dotnetHome);
+                                    Debug.Assert(process == null);
+                                    process = StartProcess(hostname, database, sqlConnectionString, Path.Combine(tempDir, benchmarksDir),
+                                        job, dotnetHome);
+                                }
 
                                 var startMonitorTime = DateTime.UtcNow;
                                 var lastMonitorTime = startMonitorTime;
                                 var oldCPUTime = TimeSpan.Zero;
 
                                 timer = new Timer(_ =>
                                 {
-                                    // TODO: Accessing the TotalProcessorTime on OSX throws so just leave it as 0 for now
-                                    // We need to dig into this
-                                    var now = DateTime.UtcNow;
-                                    var newCPUTime = OperatingSystem == OperatingSystem.OSX ? TimeSpan.Zero : process.TotalProcessorTime;
-                                    var elapsed = now.Subtract(lastMonitorTime).TotalMilliseconds;
-                                    var cpu = Math.Round((newCPUTime - oldCPUTime).TotalMilliseconds / (Environment.ProcessorCount * elapsed) * 100);
-                                    lastMonitorTime = now;
-                                    oldCPUTime = newCPUTime;
-
-                                    job.ServerCounters.Add(new ServerCounter
+                                    // If we couldn't get the lock it means one of 2 things are true:
+                                    // - We're about to dispose so we don't care to run the scan callback anyways.
+                                    // - The previous the computation took long enough that the next scan tried to run in parallel
+                                    // In either case just do nothing and end the timer callback as soon as possible
+                                    if (!Monitor.TryEnter(executionLock))
                                     {
-                                        Elapsed = now - startMonitorTime,
-                                        WorkingSet = process.WorkingSet64,
-                                        CpuPercentage = cpu
-                                    });
+                                        return;
+                                    }
 
+                                    try
+                                    {
+                                        if (disposed || Debugger.IsAttached)
```

Comment:
Why `Debugger.IsAttached`?

Reply:
Debugging this code is impossible, but I'm not sure this check is 100% required now that we have the lock.